plugin/plugin_testconn.go

```diff
@@ -0,0 +1,57 @@
+package plugin
+
+import (
+	"context"
+
+	"github.com/rs/zerolog"
+)
+
+type TestConnFailureCode string
+
+const (
+	TestConnFailureCodeUnknown            TestConnFailureCode = "UNKNOWN"
```

Comment:
Shouldn't the error codes be part of the protocol in https://github.com/cloudquery/plugin-pb/pull/22?
So we don't have to reference the SDK directly in https://github.com/cloudquery/cloudquery/pull/17993/files#diff-9f63834674c5ec7944d55a63556099b99c2ce34c0633f8bad046241910e35acbR18

Reply:
Otherwise we'd have to duplicate them in each SDK

Reply:
Empty string also falls back to UNKNOWN code, maybe we can do that? Do we want a strict list of error codes? Then we need a task force to determine the list.

Reply:
> Do we want a strict list of error codes?

Don't we already have it in the SDK? I'm suggesting the enum for the codes should be defined at the protocol level. We can still have unknown errors

Reply:
Maybe we don't need a code and only have a message

Reply:
Removed consts, keeping the code.